### PR TITLE
Introduce `PtMaxTrackCountFilter`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

To study cosmic commissioning events (especially to check detector noise), it is sometimes convenient to isolate low pT tracks.
As far as I could see, there's no such filter to retain only events with at least a track below a certain threshold. This is added here.
In addition I profit to fix the `fillDescriptions`  method of [PtMinTrackCountFilter.cc](https://github.com/cms-sw/cmssw/compare/master...mmusich:PtMinTrackCountFilter?expand=1#diff-d632b65943b2e5288d54a8e4c534126f0c96dc867790e617e87fa48cdf50eac7) which has been introduced in PR https://github.com/cms-sw/cmssw/pull/29604/ (this is done in commit: a3e107893bcd74b54f27b62e549c9365c479deac, without it, using the filter results in failure validating the configuration).

#### PR validation:

Private, tested adding:

```python
from CommonTools.RecoAlgos.ptMaxTrackCountFilter_cfi import ptMaxTrackCountFilter
process.myfilter = ptMaxTrackCountFilter.clone(src = cms.InputTag('ctfWithMaterialTracksP5'),
                                               ptMax = cms.double(3.))
```

to configuration file and tested it works correctly.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: CommonTools/RecoAlgos/plugins/PtMaxTrackCountFilter.cc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+void PtMaxTrackCountFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("src", edm::InputTag("tracks"));
+  desc.add<unsigned int>("minNumber", 1);
+  desc.add<double>("ptMax", 999.);
+  desc.add<std::string>("cut", "");
+  descriptions.add("ptMaxTrackCountFilter", desc);
+}
+
+DEFINE_FWK_MODULE(PtMaxTrackCountFilter);
```
Comment (clacaputo) [inline CommonTools/RecoAlgos/plugins/PtMaxTrackCountFilter.cc:27]: If I'm not mistaken, `PtMaxTrackCountFilter ` is not used anywhere in cmssw. Are there any plans to include it in some wf, unit test?

Comment (clacaputo) [inline CommonTools/RecoAlgos/plugins/PtMaxTrackCountFilter.cc:27]: If I'm not mistaken, `PtMaxTrackCountFilter ` is not used anywhere in cmssw. Are there any plans to include it in some wf, unit test?

**Comment by mmusich** [comment]
Hi @clacaputo, can you clarify what's holding this PR? shall unit tests be implemented here directly as a *conditio sine qua non* for accepting it?

**Comment by clacaputo** [comment]
Hi @mmusich,

>study cosmic commissioning events (especially to check detector noise), it is sometimes convenient to isolate low pT tracks

if this is critical and it's needed by the next release, we can merge the PR as it is and have the unit test in a subsequent one.
Otherwise, I would prefer to have the unit test as a part of this PR.

**Comment by mmusich** [comment]
> we can merge the PR as it is and have the unit test in a subsequent one.

Since I would push the changes exercising the new module in a different package, together with other changes (not under reco signature) I would prefer to keep it separate, unless there's a strong push (to get you an idea of where there used you can look [here](https://github.com/cms-sw/cmssw/compare/CMSSW_12_2_X...mmusich:cosmicsNoiseAnalyis?expand=1)).

**Comment by clacaputo** [comment]
+reconstruction

- PtMaxTrackCountFilter introduced for studying cosmic commissioning data
- no reco changes expected or observed
- Unit test using the filter will be added in another PR (https://github.com/cms-sw/cmssw/pull/37500#issuecomment-1099296733)

**Comment by qliphy** [comment]
please test
just to refresh

**Comment by qliphy** [comment]
+1

**Comment by mmusich** [comment]
> Unit test using the filter will be added in another PR (https://github.com/cms-sw/cmssw/pull/37500#issuecomment-1099296733)

for the record, here it is: [#37621](https://github.com/cms-sw/cmssw/pull/37621/files#diff-cafdbf2e71f011b8e79a07a30d8f54dafe829d14999556c44171189e12be3d5cR149-R151)

**Pull request merged**